src/zdns/lookup.go

```diff
@@ -897,30 +897,35 @@ func (r *Resolver) iterateOnAuthorities(ctx context.Context, qWithMeta *Question
 	if len(result.Authorities) == 0 {
```

Comment:
Let's comment `iterateOnAuthorities` (know it didn't have one before you, we'll just explain the behavior here)
Something like:

// iterateOnAuthorities takes the authorities from the referrals of a nameserver, shuffles them, and iteratively tries to do a lookup against them. If one succeeds, we return without trying the others. If one fails, we iterate to the next.